base/src/main/java/uk/ac/ebi/atlas/experimentimport/analyticsindex/AnalyticsIndexerService.java

```diff
@@ -51,18 +48,17 @@ public int index(
             while (it.hasNext()) {
                 while (addedIntoThisBatch < batchSize && it.hasNext()) {
                     SolrInputDocument analyticsInputDocument = it.next();
-                    if (analyticsIndexDocumentValidator.validate(analyticsInputDocument)) {
-                        toLoad.add(analyticsInputDocument);
-                        addedIntoThisBatch++;
-                    }
+                    toLoad.add(analyticsInputDocument);
+                    addedIntoThisBatch++;
                 }
                 if (addedIntoThisBatch > 0) {
                     UpdateResponse r = solrClient.add(toLoad);
-                    LOGGER.info("Sent {} documents for {}, qTime:{}",
+                    LOGGER.info(
+                            "Sent {} documents for {}, qTime:{}",
                             addedIntoThisBatch, experiment.getAccession(), r.getQTime());
                     addedInTotal += addedIntoThisBatch;
                     addedIntoThisBatch = 0;
-                    toLoad = new ArrayList<>(batchSize);
+                    toLoad.clear();
```

Comment:
I like this! I remember you demonstrating that this is faster than allocating the new ArrayList - the cleared elements get GC'ed immediately.